code/headers/string.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <cstdint>
+
+namespace r2d2::location {
+    /**
+     * @brief Get the offset of the separator
+     * returns 0 if no seperator is found
+     *
+     * @param string
+     * @param length
+     * @param separator
+     * @return size_t returns the length when the seperator is found in the
+     * array
+     */
+    size_t get_offset_separator(const uint8_t *string, const size_t length,
+                                const uint8_t separator);
```

Comment:
why is `const  uint8_t*` used throughout the module instead of `const char*` ? `const char*` can be used with C's `strlen()` function which eliminates the need to pass the length seperately.

Reply:
This is done so we only get a part of the string as this is not nullterminated. Strlen returns the full string lenght so that doesnt do the same thing

Reply:
The const uint8_t part is as usart.receive returns a uint8_t.